projects/angular/src/data/datagrid/datagrid-column-toggle.ts

```diff
@@ -107,7 +108,7 @@ export class ClrDatagridColumnToggle implements OnDestroy {
 
   // Without tracking the checkboxes get rerendered on model update, which leads
   // to loss of focus after checkbox toggle.
-  trackByFn: TrackByFunction<ColumnState> = index => index;
+  readonly trackByFn = columnToggleTrackByFn;
```

Comment:
Rather then ignoring the trackBy set here, could we set this trackByFn to use the same trackBy from the Items provider?

```
   constructor(
    ...
    popoverToggleService: ClrPopoverToggleService
    private items: Items,
  ) {
    if (items && items.trackBy) {
      this.trackByFn = items.trackBy;
    }
     this.subscription = popoverToggleService.openChange.subscribe(change => (this.openState = change));
  }
```

Reply:
I'm confused. We don't want to use `items.trackBy` here. This `trackByFn` is for the column checkboxes in the "manage columns" popup.

Reply:
Let’s sync on Monday about it. I’m also a bit confused on exactly what we’re solving here. The behavior we’re seeing is this track by function overrides the items trackby, so forcing them to be the same resolves that. This pr ignores the column trackby, which seems like the same thing as just removing the trackby from the columns.